pkg/broker/broker.go

```diff
@@ -880,33 +881,57 @@ func (a AnsibleBroker) Bind(instance apb.ServiceInstance, bindingUUID uuid.UUID,
 		log.Warningf("unable to retrieve provision time credentials - %v", err)
 		return nil, false, err
 	}
-	if bi, err := a.dao.GetBindInstance(bindingUUID.String()); err == nil {
-		if uuid.Equal(bi.ID, bindingInstance.ID) {
-			if reflect.DeepEqual(bi.Parameters, bindingInstance.Parameters) {
-				bindExtCreds, err := a.dao.GetExtractedCredentials(bi.ID.String())
-				if err != nil && !client.IsKeyNotFound(err) {
-					return nil, false, err
-				}
+
+	if existingBI, err := a.dao.GetBindInstance(bindingUUID.String()); err == nil {
+		if existingBI.IsEqual(bindingInstance) {
+			bindExtCreds, err := a.dao.GetExtractedCredentials(existingBI.ID.String())
+			// It's ok if there aren't any bind credentials yet.
+			if err != nil && !client.IsKeyNotFound(err) {
+				return nil, false, err
+			}
+			var createJob apb.JobState
+			if existingBI.CreateJobKey != "" {
+				createJob, err = a.dao.GetStateByKey(existingBI.CreateJobKey)
+			}
+
+			switch {
+			// unknown error
+			case err != nil && !client.IsKeyNotFound(err):
+				return nil, false, err
+			// If there is a job in "succeeded" state, or no job at all, or
+			// the referenced job no longer exists (we assume it got
+			// cleaned up eventually), assume everything is complete.
+			case createJob.State == "succeeded", existingBI.CreateJobKey == "", client.IsKeyNotFound(err):
```

Comment:
Can we use the [apb job state](https://github.com/openshift/ansible-service-broker/blob/master/pkg/apb/types.go#L184) value here?